### ISSUE TITLE
3-way binding and formly

### ISSUE THREAD

**Issue by abcd-ca** [Urigo/angular-meteor#370]
Hey, sorry if this is off topic but I'm using angular-meteor and have to do a big data input form project. I'm using Ionic Framework, FYI. I see that ng-model works for $meteor.object binding but for some reason [angular-formly](https://github.com/formly-js/angular-formly) doesn't like those kinds of objects for its model binding. Do you happen to have any insight into this?  When I bind a formly form to a $meteor.object I get,

```
TypeError: Cannot convert undefined or null to object
    at Function.keys (native)
    at _.extend.forEach
```

Thank you so much for your angular-metor tutorial, super helpful!

**Comment by Urigo** [comment]
@abcd-ca can you share a reproduction?

**Comment by Urigo** [comment]
Closing, please re-open if you have a reproduction

**Comment by fdcastel** [comment]
I made a minimal reproducible case here:

http://meteorpad.com/pad/Cr373mB7jWN8qzqzt/angular-meteor-issue-370

This same problem was encountered here:

https://github.com/formly-js/angular-formly/issues/320

**Comment by estaub** [comment]
@fdcastel See my comment on issue 320.  Consider use of AngularMeteorObject.getRawObject().

**Comment by Urigo** [comment]
@estaub could you share the link here?  I can't find it

**Comment by estaub** [comment]
@Urigo - sorry, the link to that issue was in @fdcastel 's comment:  https://github.com/formly-js/angular-formly/issues/320 
I didn't mention getRawObject() there; I found it later.
FWIW, I decided that it's best to assume that the MongoDB rendition of a model is likely to be different from that best-used by formly, so I always map between them.  That incidentally avoids the AngularMeteorObject problem.

**Comment by DrDanRyan** [comment]
I think I can provide more insight on this issue. I am using angular-meteor 1.2.3 and formly 7.3.4. If I try to use an AngularMeteorObject obtained with `$scope.$meteorObject` as a formly model, I get a ton of errors that are kicked off by:
`Error: [ng:cpws] Can't copy! Making copies of Window or Scope instances is not supported.`

However, if I use `$meteor.object` to create the object, I do not get any errors.

At some point formly must try to copy the object passed in as the model, and `$scope.$meteorObject` is somehow attaching a `$scope` to the object whereas `$meteor.object` does not....? Any thoughts?

Bottom line, you can use formly with an AngularMeteorObject as the model as long as you create it with `$meteor.object` which means that you are responsible for calling their `stop` method yourself when you are finished with the object.

**Comment by estaub** [comment]
@DrDanRyan Yes, IIRC it blew up trying to copy the scope.  However, I'd check the object returned by $meteor.object to see everything that's being copied.  While it may work, I think I recall that there was also a huge chunk of mongo that got copied with every object, making it take many times more space and time than it should.
